uberfire-runtime-plugins/uberfire-runtime-plugins-client/src/main/java/org/uberfire/ext/plugin/client/widget/navigator/PluginNavList.java

```diff
@@ -212,8 +212,10 @@ public void setup( final Collection<Plugin> plugins ) {
             //Sort Activities by Name. A TreeMap supports sorting on insertion by natural ordering of its keys
             final Map<String, Activity> activities = new TreeMap<String, Activity>( PLUGIN_NAME_COMPARATOR );
             for ( final Activity item : e.getValue() ) {
-                activities.put( item.getName(),
-                                item );
+                if ( !thereIsAlreadyAPluginWithSameName( item, activities ) ) {
```

Comment:
...and what do you do if there is a plugin with that name already? Skip it, and then Users complain plugins are "disappearing"?

IMO we either need to prevent plugins from being created with an existing name - although this is going to be close to impossible if Users can create them with either (1) using the UI, (2) using a JSON file in the webapp's folder, (3) using a JSON file committed to the plugins.git repository; or find a solution to support activities with the sane name (but IDK if this is supported by the whole Plugins framework - or whether names need to be unique?).